src/libsyntax/parse/token.rs

```diff
@@ -48,6 +48,8 @@ pub enum DelimToken {
     Bracket,
     /// A curly brace: `{` or `}`
     Brace,
+    /// An empty delimiter
+    Empty,
```

Comment:
Bikeshed: I don't think `Empty` is quite right for the name or description. Suggests to me that there is some kind of container which has nothing in it. I'd prefer something like `Implicit` or `None`.